service/dps/storm/topologies/indexing/src/main/java/eu/europeana/cloud/service/dps/storm/topologies/indexing/bolts/IndexingBolt.java

```diff
@@ -38,57 +40,67 @@ public IndexingBolt(Properties indexingProperties) {
     @Override
     public void prepare() {
         try {
-            indexerFactoryWrapper = new IndexerFactoryWrapper();
-
-        } catch (IndexerConfigurationException | URISyntaxException e) {
+            indexerPoolWrapper = new IndexerPoolWrapper(MAX_IDLE_TIME_FOR_INDEXER_IN_SECS,
+                IDLE_TIME_CHECK_INTERVAL_IN_SECS);
+        } catch (IndexingException | URISyntaxException e) {
             LOGGER.error("Unable to initialize indexer", e);
         }
     }
 
+    @Override
```

Comment:
There's no guarantee that  cleanup method will be called on the cluster:
The cleanup method is intended for when you run topologies in local mode (where a Storm cluster is simulated in process), and you want to be able to run and kill many topologies without suffering any resource leaks.